### PR TITLE
refactor: handle undefined on setAttributes params

### PR DESCRIPTION
**Pull request by NikitK-deriv**
in deriv-app we have to setAttributes several times, in this case we have to handle all undefined's properties
here is example with quick tests how it works

Screenshot 2023-10-23 at 09 42 07

### CODE REVIEW

File: src/analytics.ts [hunk 1]
```diff
@@ -30,16 +30,24 @@ export function createAnalyticsInstance(options?: Options) {
         user_id,
         app_id,
     }: TCoreAttributes) => {
+        const attributes = {
+            ...(country !== undefined && { country }),
+            ...(user_language !== undefined && { user_language }),
+            ...(device_language !== undefined && { device_language }),
+            ...(device_type !== undefined && { device_type }),
```
Comment (amir-deriv) [inline src/analytics.ts:37]:
```suggestion
            ...(country !== undefined ? { country }: {}),
            ...(user_language !== undefined ? { user_language }: {}),
```

isn't this should be like this ? performing a spread operation `...` on boolean value could throw an error

Comment (NikitK-deriv) [reply src/analytics.ts:37]: if some property is undefined, the param will not copy at all :) 
example is proof

Comment (amir-deriv) [reply src/analytics.ts:37]: Sounds good, I thought running spread operator on boolean will throw the error, but it turns out that it won't 😄

**Review by amir-deriv — APPROVED** [review]
LGTM 👍

**Comment by amir-deriv** [comment]
@NikitK-deriv can you update the version in `package.json` ?

File: src/growthbook.ts [hunk 2]
```diff
@@ -3,10 +3,10 @@ import * as RudderAnalytics from 'rudder-sdk-js'
 
 type AttributesTypes = {
     id?: string
```
Comment (ali-hosseini-deriv) [inline src/growthbook.ts:5]: based on line `44` this has to be required.

File: src/types.ts [hunk 3]
```diff
@@ -213,10 +213,10 @@ export type TEvents = {
 
 export type TCoreAttributes = {
     id?: string
```
Comment (ali-hosseini-deriv) [inline src/types.ts:215]: ditto.

**Review by ali-hosseini-deriv — APPROVED** [review]

**Pull request merged**